src/display/annotation_layer.js

```diff
@@ -717,39 +718,37 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
             }
             // Save the entered value
             elementData.userValue = event.target.value;
-            window.dispatchEvent(
-              new CustomEvent("dispatchEventInSandbox", {
+            this.linkService.eventBus?.dispatch("dispatcheventinsandbox", {
+              source: this,
+              detail: {
+                id,
+                name: "Keystroke",
+                value: event.target.value,
+                willCommit: true,
+                commitKey,
+                selStart: event.target.selectionStart,
+                selEnd: event.target.selectionEnd,
+              },
+            });
+          });
+          const _blurListener = blurListener;
+          blurListener = null;
+          element.addEventListener("blur", event => {
+            if (this._mouseState?.isDown) {
```

Comment:
`_mouseState` shoudn't be null here else the behaviour won't be correct, so you should report something (a warning or throw an exception).

Reply:
We're not currently warning/throwing for *any* of the parameters used with the `AnnotationLayer`, which assuming we even care enough to fix this (which is somewhat doubtful to me), should then probably be fixed all at once rather than for just one parameter.

Please note also that it's perfectly fine for the `mouseState` parameter to be optional in *some* cases:
 - The `AnnotationLayer` is also utilized with the viewer-components, which can be used build a simplified viewer without all the bells and whistles of the full default viewer, and in that case we *really* shouldn't require all of the scripting-related options to be provided by the user.
 - Furthermore, the *existing* viewer JSDocs specifically listed the `mouseState`-parameter as optional in e.g. the `BaseViewer` and consequently it should be optional *everywhere* including the `AnnotationLayer`.

However, I've changed things slightly to ensure that `this._mouseState` is always defined with the expected format, such that we don't need to use inline checks in (potentially) multiple places.